homeassistant/components/enphase_envoy/sensor.py

```diff
@@ -64,22 +85,25 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
             and coordinator.data.get("inverters_production") is not None
         ):
             for inverter in coordinator.data["inverters_production"]:
-                entity_name = f"{name} {sensor_description.name} {inverter}"
+                entity_name: str = f"{name} {sensor_description.name} {inverter}"
                 split_name = entity_name.split(" ")
                 serial_number = split_name[-1]
                 entities.append(
                     Envoy(
                         sensor_description,
                         entity_name,
                         name,
-                        config_entry.unique_id,
+                        cast(str, config_entry.unique_id),
```

Comment:
This cast is odd in this location. There is no guarantee the config entry has a unique ID.
Envoy could as well accept the `None` value and deal with that (which removes the need for casting)